qiskit/quantum_info/analysis/z2_symmetries.py

```diff
@@ -0,0 +1,485 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Z2Symmetries for SparsePauliOp."""
+
+from __future__ import annotations
+
+import itertools
+import logging
+from copy import deepcopy
+from typing import Optional, Union, cast
+
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info import Pauli, PauliList, SparsePauliOp, Clifford
+
+
+logger = logging.getLogger(__name__)
+
+
+class Z2Symmetries:
+    r"""
+    The $Z_2$ symmetry converter identifies symmetries from the problem hamiltonian and uses them to
+    provide a tapered - more efficient - representation of operators as Paulis for this problem. For each
+    identified symmetry, one qubit can be eliminated in the Pauli representation at the cost of having to
+    test two symmetry sectors (for the two possible eigenvalues - tapering values - of the symmetry).
+    In certain problems such as the finding of the main operator's ground state, one can a priori
+    identify the symmetry sector of the solution and thus effectively reduce the computational overhead.
+
+    The following attributes can be read and updated once the ``Z2Symmetries`` object has been
+    constructed.
+
+    Attributes:
+        tapering_values (Optional[list[int]]): Values determining the sector.
```

Comment:
```suggestion
        tapering_values (list[int] | None): Values determining the sector.
```
Not sure about this one but it would align it with my previous suggestion

Reply:
Done

Reply:
I saw this `(None or list[int])` in another Qiskit class (pauli_table.py) so I'll be using it